packages/components/src/components/buttons/NewButton/utils.ts

```diff
@@ -0,0 +1,190 @@
+            info: theme.baseContentInfoContrastInverse,
+            warning: theme.baseContentWarningContrastInverse,
+            critical: theme.baseContentNegativeContrastInverse,
+        },
+    };
+
+    return isInverse ? colorMapInverse[priority][intent] : colorMap[priority][intent];
+};
+
+export const mapPropsToCSS = (
+    intent: NewButtonIntent,
+    priority: NewButtonPriority,
+    isDisabled: boolean,
+    isInverse: boolean,
+    theme: DefaultTheme,
+): RuleSet<object> => {
+    if (isDisabled) {
+        const backgroundMapDisabled: Record<NewButtonPriority, CSSColor> = {
+            primary: theme.stateFillElementBoldDisabled,
+            secondary: theme.stateFillElementSoftDisabled,
+        };
+
+        const backgroundMapDisabledInverse: Record<NewButtonPriority, CSSColor> = {
+            primary: theme.stateFillElementBoldInverseDisabled,
+            secondary: theme.stateFillElementSoftInverseDisabled,
+        };
+
+        const backgroundColor = isInverse
+            ? backgroundMapDisabledInverse[priority]
+            : backgroundMapDisabled[priority];
+
+        return css`
+            background: ${backgroundColor};
+        `;
+    }
+
+    const backgroundMap: Record<NewButtonPriority, Record<NewButtonIntent, CSSColor>> = {
+        primary: {
+            brand: theme.baseFillElementBrandBold,
+            neutral: theme.baseFillElementContrast,
+            info: theme.baseFillElementInfoBold,
+            warning: theme.baseFillElementWarningBold,
+            critical: theme.baseFillElementNegativeBold,
+        },
+        secondary: {
+            brand: theme.baseFillElementBrandSoft,
+            neutral: theme.baseFillElementNeutralSoft,
+            info: theme.baseFillElementInfoSoft,
+            warning: theme.baseFillElementWarningSoft,
+            critical: theme.baseFillElementNegativeSoft,
+        },
+    };
+
+    const backgroundMapInverse: Record<NewButtonPriority, Record<NewButtonIntent, CSSColor>> = {
+        primary: {
+            brand: theme.baseFillElementBrandBoldInverse,
+            neutral: theme.baseFillElementNeutralLight,
+            info: theme.baseFillElementInfoBoldInverse,
+            warning: theme.baseFillElementWarningBoldInverse,
+            critical: theme.baseFillElementNegativeBoldInverse,
+        },
+        secondary: {
+            brand: theme.baseFillElementBrandSoftInverse,
+            neutral: theme.baseFillElementNeutralSoftInverse,
+            info: theme.baseFillElementInfoSoftInverse,
+            warning: theme.baseFillElementWarningSoftInverse,
+            critical: theme.baseFillElementNegativeSoftInverse,
+        },
+    };
+
+    const backgroundColor = isInverse
+        ? backgroundMapInverse[priority][intent]
+        : backgroundMap[priority][intent];
+
+    return css`
+        background: ${backgroundColor};
+
+        &:hover {
+            background: ${addAlphaToHex(backgroundColor, 82)};
```

Comment:
shouldn't we put these numbers as const? We might use it also for other components